From 784184bc012ac249785349eb66689abfc3cff35a Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Wed, 8 Nov 2017 20:54:41 +0100
Subject: [PATCH 1/1] scripts/ld-version.sh: regular expression compile fails

ls --version | scripts/ld-version.sh
fails with
awk: scripts/ld-version.sh:
line 4: regular expression compile failed (missing '(')
.*)

So let's refresh the script with the current Linux kernel
version.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 scripts/ld-version.sh | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/scripts/ld-version.sh b/scripts/ld-version.sh
index 198580d245..d135882e2c 100755
--- a/scripts/ld-version.sh
+++ b/scripts/ld-version.sh
@@ -1,8 +1,10 @@
 #!/usr/bin/awk -f
 # extract linker version number from stdin and turn into single number
 	{
-	gsub(".*)", "");
+	gsub(".*\\)", "");
+	gsub(".*version ", "");
+	gsub("-.*", "");
 	split($1,a, ".");
-	print a[1]*10000000 + a[2]*100000 + a[3]*10000 + a[4]*100 + a[5];
+	print a[1]*100000000 + a[2]*1000000 + a[3]*10000;
 	exit
 	}
-- 
2.14.2

